x/auth/mapper.go

```diff
@@ -116,6 +130,26 @@ func (am AccountMapper) setSequence(ctx sdk.Context, addr sdk.Address, newSequen
 	return nil
 }
 
+// Returns and increments the global account number counter
+func (am AccountMapper) GetNextAccountNumber(ctx sdk.Context) int64 {
+	var accNumber int64
+	store := ctx.KVStore(am.key)
+	bz := store.Get(globalAccountNumberKey)
+	if bz == nil {
+		accNumber = 0
+	} else {
+		err := am.cdc.UnmarshalBinary(bz, &accNumber)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	bz = am.cdc.MustMarshalBinary(accNumber + 1)
```

Comment:
Overflow concerns?

Reply:
It's an int64.  Do we really have to worry about overflowing that?

Reply:
Max int64 = `9223372036854775807`, assuming 1s blocks and 1000 `NewAccount` calls per block, it would take `106751991167` days - or `300,000` millenia - to overflow. 

OK, probably not.

Reply:
Hopefully Go implements int128 by then :crossed_fingers: